datasette/app.py

```diff
@@ -443,6 +443,37 @@ def __init__(
         self._root_token = secrets.token_hex(32)
         self.client = DatasetteClient(self)
 
+    async def apply_metadata_json(self):
+        # Apply any metadata entries from metadata.json to the internal tables
+        # step 1: top-level metadata
+        for key in self._metadata_local or {}:
```

Comment:
Maybe we should rename `self._metadata_local` to `self._metadata_on_startup` or `self._metadata_from_file` to help clarify that it's the metadata that was passed to Datasette's constructor on startup?

I don't want people to get confused and think that reading from `self._metadata_local` is a useful thing to do, when they should be calling the new async methods instead.

Probably overthinking that though, since those methods will be documented but `self._metadata_local` will not.